lib/logstash/pluginmanager/install.rb

```diff
@@ -1,82 +1,128 @@
 require 'clamp'
 require 'logstash/namespace'
 require 'logstash/environment'
-require 'logstash/pluginmanager'
 require 'logstash/pluginmanager/util'
-require 'rubygems/dependency_installer'
-require 'rubygems/uninstaller'
 require 'jar-dependencies'
 require 'jar_install_post_install_hook'
 require 'file-dependencies/gem'
 
+require "logstash/gemfile"
+require "logstash/bundler"
+
 class LogStash::PluginManager::Install < Clamp::Command
+  parameter "[PLUGIN] ...", "plugin name(s) or file"
+  option "--version", "VERSION", "version of the plugin to install"
+  option "--force", :flag, "force install without verifying plugin validity"
+  option "--development", :flag, "install all development dependencies of currently installed plugins"
 
-  parameter "PLUGIN", "plugin name or file"
+  # the install logic below support installing multiple plugins with each a version specification
+  # but the argument parsing does not support it for now so currently if specifying --version only
+  # one plugin name can be also specified.
+  #
+  # TODO: find right syntax to allow specifying list of plugins with optional version specification for each
 
-  option "--version", "VERSION", "version of the plugin to install", :default => ">= 0"
+  def execute
+    if development?
+      unless plugin_list.empty?
+        $stderr.puts("Cannot specify plugin(s) with --development, it will add the development dependencies of the currently installed plugins")
+        exit(1)
+      end
+    else
+      if plugin_list.empty? && !force?
+        $stderr.puts("No plugin specified")
+        exit(1)
+      end
+      if version && plugin_list.size > 1
+        # temporary until we fullfil TODO ^^
+        $stderr.puts("Only 1 plugin name can be specified with --version")
+        exit(1)
+      end
+    end
 
-  option "--proxy", "PROXY", "Use HTTP proxy for remote operations"
+    unless File.writable?(LogStash::Environment::GEMFILE_PATH)
+      $stderr.puts("File #{LogStash::Environment::GEMFILE_PATH} does not exist or is not writable, aborting")
+      exit(1)
+    end
 
-  def execute
-    LogStash::Environment.load_logstash_gemspec!
+    gemfile = LogStash::Gemfile.new(File.new(LogStash::Environment::GEMFILE_PATH, "r+")).load
+    # keep a copy of the gemset to revert on error
+    original_gemset = gemfile.gemset.copy
 
-    ::Gem.configuration.verbose = false
-    ::Gem.configuration[:http_proxy] = proxy 
+    # force Rubygems sources to our Gemfile sources
+    Gem.sources = gemfile.gemset.sources
 
-    puts ("validating #{plugin} #{version}")
+    # install_list will be an array of [plugin name, version] tuples, version can be nil
+    install_list = []
```

Comment:
was really meant for clarity. despite what you may believe since our "explicit return" argument ;) I sometimes prefer to add somewhat useless but explicit constructs so it is easier to understand